src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -371,6 +374,108 @@ def image_mask_iterator():
     return nwbfile
 
 
+def add_fluorescence_traces(
+    segmentation_extractor: SegmentationExtractor,
+    nwbfile: NWBFile,
+    metadata: Optional[dict],
+    plane_index: int = 0,
+) -> NWBFile:
+    """
+    Adds the fluorescence traces specified by the metadata to the nwb file.
+    The fluorescence traces that are added are the one located in metadata["Ophys"]["Fluorescence"]
+
+    Parameters
+    ----------
+    segmentation_extractor : SegmentationExtractor
+        The segmentation extractor to get the traces from.
+    nwbfile : NWBFile
+        The nwbfile to add the fluorescence traces to.
+    metadata : dict
+        The metadata for the fluorescence traces.
+    plane_index : int, optional
+        The index of the plane to add the fluorescence traces to.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence traces added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    ophys = get_module(nwbfile, "ophys")
+
+    add_plane_segmentation(segmentation_extractor=segmentation_extractor, nwbfile=nwbfile, metadata=metadata)
+
+    # Get plane segmentation from the image segmentation
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    image_segmentation_name = image_segmentation_metadata["name"]
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    plane_segmentation_name = image_segmentation_metadata["plane_segmentations"][0]["name"]
+    plane_segmentation = image_segmentation.plane_segmentations[plane_segmentation_name]
+
+    # Create a reference for ROIs from the plane segmentation
+    roi_table_region = plane_segmentation.create_roi_table_region(
+        region=segmentation_extractor.get_roi_ids(),
+        description=f"region for Imaging plane{plane_index}",
+    )
+
+    # Create fluorescence container
+    if fluorescence_name in ophys.data_interfaces:
+        fluorescence = ophys.get_data_interface(fluorescence_name)
+    else:
+        fluorescence = Fluorescence(name=fluorescence_name)
+        # Add fluorescence to the ophys module
+        ophys.add(fluorescence)
+
+    # Get traces from the segmentation extractor
+    roi_response_dict = segmentation_extractor.get_traces_dict()
```

Comment:
open for discussion:
`get_traces_dict()` returns the `dff` signal if exists,
to add `dff` signal to the `DfOverF` object instead of `Fluorescence` container, lines from 452 to 474 should be refactored in a way e.g.:
- create a private method that returns a single roi response series based on its name --> then this can be added to a `DfOverF` object